sdccc/src/main/java/com/draeger/medical/sdccc/tests/biceps/invariant/InvariantMessageModelAnnexTest.java

```diff
@@ -241,6 +241,47 @@ void testRequirementC5() throws NoTestData, IOException {
                 acceptableSequenceSeen.get(), "No DescriptionModificationReport seen during test run, test failed.");
     }
 
+    @Test
+    @TestIdentifier(EnabledTestConfig.BICEPS_C7)
+    @TestDescription("Checks all DescriptionModificationReports seen during the TestRun for ReportParts whose "
+            + "./Descriptor references an MdsDescriptor and ensures that these ReportParts do not have the "
+            + "@ParentDescriptor attribute set.")
+    @RequirePrecondition(
+            simplePreconditions = ConditionalPreconditions.DescriptionModificationMdsDescriptorPrecondition.class)
+    void testRequirementC7() throws NoTestData, IOException, MarshallingException {
+        final var acceptableReportsSeen = new AtomicInteger(0);
+
+        try (final MessageStorage.GetterResult<MessageContent> descriptionModificationReports =
+                 messageStorage.getInboundMessagesByBodyType(Constants.MSG_DESCRIPTION_MODIFICATION_REPORT)) {
+            for (MessageContent messageContent :
+                descriptionModificationReports.getStream().toList()) {
+                final SoapMessage soapMessage = marshalling.unmarshal(
+                    new ByteArrayInputStream(messageContent.getBody().getBytes(StandardCharsets.UTF_8)));
+                final DescriptionModificationReport descriptionModificationReport = (DescriptionModificationReport)
+                    soapMessage.getOriginalEnvelope().getBody().getAny().get(0);
```

Comment:
You can use SoapUtil to extract the DescriptionModificationReport

Reply:
Thanks & Done.